src/core/lender/assessor/default.sol

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2019 Centrifuge
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "./base.sol";
+import "tinlake-math/interest.sol";
+
+contract DefaultAssessor is BaseAssessor, Interest {
+
+    // accrueTrancheInterest can implement different interest models
+    function accrueTrancheInterest(address tranche_) public returns (uint) {
+        SeniorTrancheLike tranche = SeniorTrancheLike(tranche_);
+        uint debt = tranche.borrowed() + tranche.interest();
+        // move to tinlake-math
+        // interest is calculated based on tranche debt
+        return safeSub(rmul(rpow(tranche.ratePerSecond(), now - tranche.lastUpdated(), ONE), debt), debt);
```

Comment:
if we update the naming in tinlake-math, we can directly use `safeSub(updateChi(ONE, ratePerSecond, lastUpdated), debt)`

Reply:
yes. we need to make the method more general.

It is not updateChi anymore.

Reply:
yes, exactly